onnxruntime/core/providers/qnn/qnn_allocator.cc

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/qnn/qnn_allocator.h"
+
+#include <cassert>
+#include <cstddef>
+#include <algorithm>
+#include <limits>
+
+#include "core/common/common.h"
+#include "core/mlas/inc/mlas.h"  // for MlasGetPreferredBufferAlignment()
+
+namespace onnxruntime::qnn {
+
+namespace {
+
+struct AllocationHeader {
```

Comment:
Would be great to add a comment describing the overall setup and how it uses this header.

Reply:
added some description, let me know if there's any other info that'd be useful to add.